pkg/operator/apis/aro.openshift.io/v1alpha1/cluster_types.go

```diff
@@ -95,6 +95,59 @@ type ClusterList struct {
 	Items           []Cluster `json:"items"`
 }
 
+// ConditionStatus indicates the status of a condition (true, false, or unknown).
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition;
+// "ConditionFalse" means a resource is not in the condition; "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+// ConditionType is a valid value for APIServiceCondition.Type
+type ConditionType string
+
+const (
+	// Available indicates that the service exists and is reachable
+	Available ConditionType = "Available"
+)
+
+// Condition represents an observation of an object's
+// state. Conditions are an extension mechanism intended to be used
+// when the details of an observation are not a priori known or would
+// not apply to all instances of a given Kind. \n Conditions should
+// be added to explicitly convey properties that users and components
+// care about rather than requiring those properties to be inferred
+// from other observations. Once defined, the meaning of a Condition
+// can not be changed arbitrarily - it becomes part of the API, and
+// has the same backwards- and forwards-compatibility concerns of
+// any other part of the API.
+type Condition struct {
```

Comment:
and here https://github.com/Azure/ARO-RP/blob/master/vendor/k8s.io/api/core/v1/types.go#L2543-L2563